src/proof/proof_rule.h

```diff
@@ -1073,171 +1073,277 @@ enum class PfRule : uint32_t
+   *
+   * where :math:`C` is the :math:`n^{\mathit{th}}` constructor of the type of
+   * t, and :math:`\mathit{is}_C` is the discriminator (tester) for :math:`C`.
+   * \endverbatim
+   */
   DT_INST,
-  // ======== Collapse
-  // Children: none
-  // Arguments: ((sel_i (C_j t_1 ... t_n)))
-  // ----------------------------------------
-  // Conclusion: (= (sel_i (C_j t_1 ... t_n)) r)
-  // where C_j is a constructor, r is t_i if sel_i is a correctly applied
-  // selector, or TypeNode::mkGroundTerm() of the proper type otherwise. Notice
-  // that the use of mkGroundTerm differs from the rewriter which uses
-  // mkGroundValue in this case.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Datatypes -- Collapse**
+   *
+   * .. math::
+   *
+   *   \inferrule{-\mid \mathit{sel}_i(C_j(t_1,\dots,t_n))}{
+   *   \mathit{sel}_i(C_j(t_1,\dots,t_n)) = r}
+   *
+   * where :math:`C_j` is a constructor, :math:`r` is :math:`t_i` if
+   * :math:`\mathit{sel}_i` is a correctly applied selector, or
+   * ``TypeNode::mkGroundTerm()`` of the proper type otherwise. Notice that the
+   * use of ``mkGroundTerm`` differs from the rewriter which uses
+   * ``mkGroundValue`` in this case.
+   * \endverbatim
+   */
   DT_COLLAPSE,
-  // ======== Split
-  // Children: none
-  // Arguments: (t)
-  // ----------------------------------------
-  // Conclusion: (or ((_ is C1) t) ... ((_ is Cn) t))
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Datatypes -- Split**
+   *
+   * .. math::
+   *
+   *   \inferrule{-\mid t}{\mathit{is}_{C_1}(t)\vee\cdots\vee\mathit{is}_{C_n}(t)}
+   *
+   * where :math:`C_1,\dots,C_n` are all the constructors of the type of :math:`t`.
+   * \endverbatim
+   */
   DT_SPLIT,
-  // ======== Clash
-  // Children: (P1:((_ is Ci) t), P2: ((_ is Cj) t))
-  // Arguments: none
-  // ----------------------------------------
-  // Conclusion: false
-  // for i != j.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Datatypes -- Clash**
+   *
+   * .. math::
+   *
+   *   \inferruleSC{\mathit{is}_{C_i}(t), \mathit{is}_{C_j}(t)\mid -}{\bot}
+   *   {if $i\neq j$}
+   *
+   * \endverbatim
+   */
   DT_CLASH,
 
-  //================================================= Quantifiers rules
-  // ======== Skolem intro
-  // Children: none
-  // Arguments: (k)
-  // ----------------------------------------
-  // Conclusion: (= k t)
-  // where t is the original form of skolem k.
+  /**
+   * \verbatim embed:rst:leading-asterisk
+   * **Quantifiers -- Skolem introduction**
+   *
+   * .. math::
+   *
+   *   \inferrule{-\mid k}{k = t}
+   *
+   * where :math:`t` is the original form of skolem :math:`k`.
+   * \endverbatim
+   */
   SKOLEM_INTRO,
-  // ======== Exists intro
-  // Children: (P:F[t])
-  // Arguments: ((exists ((x T)) F[x]))
-  // ----------------------------------------
-  // Conclusion: (exists ((x T)) F[x])
-  // This rule verifies that F[x] indeed matches F[t] with a substitution
-  // over x.
+  /**
```

Comment:
Note this rule is no longer necessary, maybe we should just delete it now.

Reply:
Might as well. Will do.